File: common/src/main/java/io/xol/chunkstories/util/OSHelper.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.util;

public class OSHelper {
	public static String getOS() {
		String OS = System.getProperty("os.name").toLowerCase();
		if (OS.contains("windows")) {
			return "windows";
		}
		if (OS.contains("mac")) {
			return "macosx";
		}
		if (OS.contains("linux") || OS.contains("unix")) {
			return "linux";
		}
		if (OS.contains("bsd")) {
			return "freebsd";
		}
		if (OS.contains("sunos")) {
			return "solaris";
		}
		return "unknown";
	}

	public static boolean isWindows() {
		return getOS().equals("windows");
	}

	public static boolean isMac() {
		return getOS().equals("macosx");
	}

	public static boolean isLinux() {
		return getOS().equals("linux");
	}
}

File: common/src/main/java/xyz/chunkstories/util/OSHelper.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.util

/** For loading platform-specific dlls and stuff */
enum class SupportedOS {
    WINDOWS,
    LINUX,
    OSX,
}

object OSHelper {
    val os: SupportedOS

    init {
        val osString = System.getProperty("os.name").toLowerCase()
        os = when {
            osString.contains("windows") -> SupportedOS.WINDOWS
            osString.contains("mac") -> SupportedOS.OSX
            osString.contains("linux") -> SupportedOS.LINUX
            osString.contains("bsd") || osString.contains("unix") || osString.contains("sunos") -> throw Exception("Unsupported os: $osString")
            else -> throw Exception("Unreognised os; $osString")
        }
    }

}
